ui/src/time_picker.tsx

```diff
@@ -0,0 +1,269 @@
+type TextInputProps = {
+  props: {
+    onClick?: React.MouseEventHandler<HTMLInputElement | HTMLTextAreaElement>,
+    onChange?: React.FormEventHandler<HTMLInputElement | HTMLTextAreaElement>,
+    value?: Date | null,
+    placeholder?: S,
+    label?: S,
+    required?: B,
+    hour_cycle?: S,
+    errorMsg?: S,
+    error?: B,
+    disabled?: B
+  }
+}
+
+const
+  ThemeProvider = React.lazy(() => import('@mui/material/styles').then(({ ThemeProvider }) => ({ default: ThemeProvider }))),
+  LocalizationProvider = React.lazy(() => import('@mui/x-date-pickers').then(({ LocalizationProvider }) => ({ default: LocalizationProvider }))),
+  TimePicker = React.lazy(() => import('@mui/x-date-pickers').then(({ TimePicker }) => ({ default: TimePicker }))),
+  parseTimeStringToDate = (time: S) => new Date(`2000-01-01T${time.slice(0, 5)}:00`),
+  formatDateToTimeString = (date: Date, hour_cycle: S = '12') => date.toLocaleString('en-US', { hour: '2-digit', minute: '2-digit', hourCycle: hour_cycle === '12' ? 'h12' : 'h23' }),
+  LazyLoadPlaceholder = () =>
+    <div data-test='lazyload' style={{ height: 59 }}>
+      <Fluent.Spinner styles={{ root: { height: '100%' } }} size={Fluent.SpinnerSize.small} />
+    </div>,
+  Toolbar = ({ params: { parsedValue, setOpenView, ampm }, label, switchAmPm }: ToolbarProps) => {
+    const time = parsedValue ? formatDateToTimeString(parsedValue, ampm ? '12' : '24') : parsedValue
+    return <div className={css.toolbar}>
+      {label && <Fluent.Label className={css.toolbarLabel}>{label}</Fluent.Label>}
+      <Fluent.Text className={css.toolbarText}>
+        <Fluent.Text className={css.toolbarTime} onClick={() => setOpenView('hours')}>{time?.substring(0, 2) || '--'}</Fluent.Text>{':'}
+        <Fluent.Text className={css.toolbarTime} onClick={() => setOpenView('minutes')}>{time?.substring(3, 5) || '--'}</Fluent.Text>{' '}
+        <Fluent.Text className={css.toolbarTime} onClick={switchAmPm}>{time?.substring(6, 8) || ''}</Fluent.Text>
+      </Fluent.Text>
+    </div>
+
+  },
+  TextInput = ({ props: { onClick, onChange, placeholder, value, label, required, hour_cycle = '12', errorMsg, error, disabled } }: TextInputProps) =>
+    <Fluent.TextField
+      iconProps={{ iconName: 'Clock', }}
+      onClick={onClick}
+      onChange={onChange}
+      placeholder={placeholder || 'Select a time'}
+      disabled={disabled}
+      readOnly={true}
+      value={value ? formatDateToTimeString(value, hour_cycle) : ''}
+      label={label}
+      required={required}
+      styles={{ field: { cursor: 'pointer' }, icon: { bottom: 7 } }}
+      errorMessage={error ? errorMsg : undefined}
+    />
+
+
+export const
+  XTimePicker = ({ model: m }: { model: TimePicker }) => {
+    const
+      { hour_cycle = '12' } = m,
+      defaultVal = m.value ? parseTimeStringToDate(m.value,) : null,
+      [value, setValue] = React.useState(defaultVal),
```

Comment:
Is defaultVal really necessary? If not:
```suggestion
      [value, setValue] = React.useState(m.value ? parseTimeStringToDate(m.value) : null),
```